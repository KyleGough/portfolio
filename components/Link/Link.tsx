import NextLink from 'next/link';
import React from 'react';

interface LinkProps {
  ariaLabel?: string;
  children: React.ReactNode;
  className?: string;
  href?: string;
  onClick?: () => void;
  to?: string;
}

export const Link: React.FC<LinkProps> = ({
  className = 'text-link hover:text-link-hover focus:text-link-hover',
  to,
  href,
  onClick,
  ariaLabel,
  children,
}) => {
  return (
    <>
      {to ? (
        <NextLink className={className} href={to} onClick={onClick}>
          {children}
        </NextLink>
      ) : (
        <a
          className={className}
          href={href}
          target="_blank"
          rel="noopener noreferrer"
          onClick={onClick}
          aria-label={ariaLabel}
        >
          {children}
        </a>
      )}
    </>
  );
};
