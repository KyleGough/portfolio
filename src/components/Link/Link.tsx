import React from 'react';
import { Link as RouterLink } from 'react-router-dom';

interface LinkProps {
  to?: string;
  href?: string;
  children: React.ReactNode;
}

export const Link: React.FC<LinkProps> = ({ to, href, children }) => {
  return (
    <>
      {to ? (
        <RouterLink
          className="text-link hover:text-link-hover focus:text-link-hover"
          to={to}
        >
          {children}
        </RouterLink>
      ) : (
        <a
          className="text-link hover:text-link-hover focus:text-link-hover"
          href={href}
          target="_blank"
          rel="noopener noreferrer"
        >
          {children}
        </a>
      )}
    </>
  );
};
