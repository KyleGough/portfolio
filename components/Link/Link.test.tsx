import { render, screen } from '@testing-library/react';
import React from 'react';

import { Link } from './Link';

describe('Link component', () => {
  it('renders external link', () => {
    const { container } = render(<Link href="https://google.com">Link</Link>);

    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      'https://google.com'
    );
    expect(screen.getByText('Link')).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it('renders internal link', () => {
    render(<Link to="/test">Link</Link>);

    expect(screen.getByRole('link')).toHaveAttribute('href', '/test');
    expect(screen.getByText('Link')).toBeVisible();
  });
});
