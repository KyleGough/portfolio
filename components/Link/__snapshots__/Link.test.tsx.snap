// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders external link 1`] = `
<div>
  <a
    class="text-link hover:text-link-hover focus:text-link-hover"
    href="https://google.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    Link
  </a>
</div>
`;
